[DO NOT MERGE] Add Ai Auto-Collection to Benchmarks

Dmitry-Matveev opened the pull request:
PR to add AI ASP.NET auto-collection to Benchmarks app.
This PR can be used to compare benchmarks with/without AI (based on `EnableAi` setting) from the platform perspective.

The Instrumentation Key provided (in `AiInstrumentationKey` setting) is a temporary AI resource provisioned specifically for this purpose. We can think of changing this IKey / removing this resource after the test or switch to reading it from the environment variable to avoid exposing the IKey.

I do not expect that this PR is to be merged unless @sebastienros would think otherwise.

P.S. I validated that AI collects telemetry with this app if I run Benchmarks.dll locally in Kestrel/Sockets.

dnfclas commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/aspnet/Benchmarks?pullRequest=538) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/aspnet/Benchmarks?pullRequest=538) before we can accept your contribution.<br/><br/>:x: Dmitry-Matveev [sign now](https://cla.dotnetfoundation.org/aspnet/Benchmarks?pullRequest=538)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/aspnet/Benchmarks?pullRequest=538) it.</sub>

davidfowl commented:
What this for?

Eilon commented:
I think the idea is to see what impact there is for apps when AppInsights is enabled.

sebastienros commented on src/Benchmarks/appsettings.json:
```diff
@@ -1,3 +1,5 @@
-  "ConnectionString": "Server=(localdb)\\mssqllocaldb;Database=aspnet5-Benchmarks;Trusted_Connection=True;MultipleActiveResultSets=true"
+  "ConnectionString": "Server=(localdb)\\mssqllocaldb;Database=aspnet5-Benchmarks;Trusted_Connection=True;MultipleActiveResultSets=true",
+  "EnableAI": true,
+  "AiInstrumenationKey": "6a70388f-9193-463b-9117-4a88ba0e04c6"
```
You can remove these, we can pass them by command line args, as it's a source for the app settings. This way we can keep it secret and change it at will.

sebastienros commented on src/Benchmarks/Configuration/AppSettings.cs:
```diff
@@ -5,6 +5,10 @@ namespace Benchmarks.Configuration
 {
     public class AppSettings
     {
+        public string AiInstrumenationKey { get; set; }
```
Small typo (won't matter for the tests)

Dmitry-Matveev commented:
@sebastienros , addressed PR comments, thanks!

@davidfowl , this PR is to temporary add AI with sending ~5 telemetry items per second (unlike #361 where sample rate is 0) to measure AI's impact if it's considered "part of the platform". I would expect some heavy overhead on Benchmarks because currently App is doing mostly nothing and AI code will essentially become app's logic code. I would not think we need to merge this PR if we can measure perf off this PR commit, but you guys may think otherwise. Let me know if you need some more tuning in PR.

Dmitry-Matveev commented:
@sebastienros , please let me know if something else is needed in this PR to allow a perf run off this PR with AI enabled with some actual IKey. I haven't signed CLA yet to make sure it's not accidentally merged unless ASP.NET Core team would think it's the right thing to do.
